src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -240,6 +240,33 @@ install_packages (RpmOstreeTreeComposeContext  *self,
     if (!rpmostree_context_setup (self->corectx, tmprootfs_abspath, NULL, self->treespec,
                                   cancellable, error))
       return FALSE;
+
+#define TMP_SELINUX_ROOTFS "selinux.tmp/etc/selinux"
+
+    /* By default, the core starts with the SELinux policy of the root, but if we have a
+     * previous commit, it's much likelier that its policy will be closer to the final
+     * policy than the host system's policy. And in the case they match, we skip a full
+     * relabeling phase. Let's use that instead. */
+    if (self->previous_checksum)
```

Comment:
Let's also only do this if `selinux = true` in the treefile.

Reply:
Good point, done! :arrow_down: